### PR TITLE
Feature/user data

### PR DESCRIPTION
**Pull request by PeacefulPieDev**
While I was implementing UserData I find that lib with country and currency codes isn't really good enough. It looks too complicated for such a simple purpose. I think we should re-implement it by ourselves and rework UserData and country and currency codes.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ android {
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile 'com.android.support:support-v4:23.4.0'
-    compile 'com.google.android.gms:play-services-vision:17.0.2'
+    compile 'com.google.android.gms:play-services-vision:9.4.0+'
```
Comment (oleksandr-bilous) [inline app/build.gradle]: Why did you changed it? 9.4.0 version of library works awful and doesn't recognize a text that the 17.0.2 does.

File: app/src/main/java/currencyConverter/codes/CodeUtils.java [hunk 2]
```diff
@@ -22,17 +22,31 @@ public static CurrencyCode getCurrencyFromCountry(CountryCode countryCode) throw
...
+
+	public static List<CountryCode> getAvailableCountryCodes(boolean sortByName) {
+		ArrayList<CountryCode> codeList = new ArrayList<CountryCode>();
+
+		for (com.neovisionaries.i18n.CountryCode country : com.neovisionaries.i18n.CountryCode.values()) {
+			codeList.add(new CountryCode(country.getAlpha2()));
+		}
+
+//	    if(sortByName) {
+//		    Collections.sort(codeList);
```
Comment (oleksandr-bilous) [inline app/src/main/java/currencyConverter/codes/CodeUtils.java]: Are you sure we do not need commented code?

File: app/src/main/java/currencyConverter/codes/CodeUtils.java [hunk 3]
```diff
@@ -22,17 +22,31 @@ public static CurrencyCode getCurrencyFromCountry(CountryCode countryCode) throw
...
 	    for (com.neovisionaries.i18n.CurrencyCode currency : com.neovisionaries.i18n.CurrencyCode.values()) {
-		    codeList.add(currency.getName());
+		    codeList.add(new CurrencyCode(currency));
 	    }
 
-	    if(sortByName) {
-		    Collections.sort(codeList);
-	    }
+//	    if(sortByName) {
+//		    Collections.sort(codeList);
```
Comment (oleksandr-bilous) [inline app/src/main/java/currencyConverter/codes/CodeUtils.java]: Are you sure we do not need commented code?

**Review by oleksandr-bilous — CHANGES_REQUESTED** [review]

File: app/src/main/java/currencyConverter/codes/CodeUtils.java (hunk 2 above)
Comment (PeacefulPieDev) [reply app/src/main/java/currencyConverter/codes/CodeUtils.java]: Removed. Now I'm using sorting in array adapter.

File: app/build.gradle (hunk 1 above)
Comment (PeacefulPieDev) [reply app/build.gradle]: Reverted.

File: app/src/main/java/currencyConverter/codes/CodeUtils.java (hunk 3 above)
Comment (PeacefulPieDev) [reply app/src/main/java/currencyConverter/codes/CodeUtils.java]: Removed. Now I'm using sorting in array adapter.

**Review by oleksandr-bilous — APPROVED** [review]

**Pull request merged**